src/journal/bsod.c

```diff
@@ -0,0 +1,199 @@
+
+        xsprintf(boot_id_filter, "_BOOT_ID=" SD_ID128_FORMAT_STR, SD_ID128_FORMAT_VAL(boot_id));
+
+        r = sd_journal_add_match(j, boot_id_filter, 0);
+        if (r < 0)
+                return log_warning_errno(r, "Failed to add boot ID filter: %m");
+
+        r = sd_journal_add_match(j, "_UID=0", 0);
+        if (r < 0)
+                return log_warning_errno(r, "Failed to add User ID filter: %m");
+
+        assert_cc(0 == LOG_EMERG);
+        r = sd_journal_add_match(j, "PRIORITY=0", 0);
+        if (r < 0)
+                return log_warning_errno(r, "Failed to add Emergency filter: %m");
+
+        r = sd_journal_seek_head(j);
+        if (r < 0)
+                return log_error_errno(r, "Failed to seek to start of jornal: %m");
+
+        r = sd_journal_next(j);
+        if (r < 0)
+                return log_error_errno(r, "Failed to read next journal entry: %m");
+        if (r == 0) {
+                log_debug("No entries in the journal");
+                return r;
+        }
+
+        r = sd_journal_get_data(j, "MESSAGE", &d, &l);
+        if (r < 0)
+                return log_error_errno(r, "Failed to read journal message: %m");
+
+        message = strndup((char *) d, l);
+        if (! *ret)
+                return log_oom();
+
+        *ret = TAKE_PTR(message);
+
+        return r;
+}
+
+static int find_next_free_vt(int fd) {
+        size_t i;
+        struct vt_stat terminal_status;
+
+        if (ioctl(fd, VT_GETSTATE, &terminal_status) < 0)
+                return -errno;
+
+        for (i = 0; i < sizeof(terminal_status.v_state) * 8; i++) {
+                if ((terminal_status.v_state & (1 << i)) == 0)
+                        break;
+        }
+        return i;
+}
+
+static int display_emergency_message_fullscreen(char * message) {
+        assert(message);
+        int r;
+        char tty[STRLEN("/dev/tty") + DECIMAL_STR_MAX(int) + 1];
+        _cleanup_close_ int fd = -EBADF;
+
+        fd = open_terminal("/dev/tty1", O_RDWR|O_NOCTTY|O_CLOEXEC);
+        if (fd < 0)
+                return log_error_errno(fd, "Failed to open tty1: %m");
+
+        r = find_next_free_vt(fd);
+        if (r < 0)
+                return log_error_errno(r, "Failed to find a free VT: %m");
+
+        xsprintf(tty, "/dev/tty%d", r + 1);
+
+        fd = open_terminal(tty, O_RDWR|O_NOCTTY|O_CLOEXEC);
+        if (fd < 0)
+                return log_error_errno(fd, "Failed to open tty: %m");
+
+        if (ioctl(fd, VT_ACTIVATE, r + 1) < 0)
+                return log_error_errno(errno, "Failed to activate tty: %m");
+
+        r = loop_write(fd, ANSI_HOME_CLEAR, strlen(ANSI_HOME_CLEAR), false);
+        if (r < 0)
+                log_warning_errno(r, "Failed to clear terminal, ignoring: %m");
+
+        r = loop_write(fd, message, strlen(message), false);
+        if (r < 0)
+                return log_warning_errno(r, "Failed to write emergency message to terminal: %m");
+
```

Comment:
please wait for a key press here, and if you receive it, switch back to the original VT